--- a/haxe/zip/Uncompress.unit.hx
+++ b/haxe/zip/Uncompress.unit.hx
@@ -1,5 +1,5 @@
 // not supported in js/python/cs yet
-#if (cpp || php || java || neko || flash)
+#if (cpp || php || java || neko || flash || ruby)
 var d = [120, 218, 43, 73, 45, 46, 1, 0, 4, 93, 1, 193];
 var b = haxe.io.Bytes.alloc(d.length);
 for (i in 0...d.length)
@@ -13,7 +13,7 @@ for (i in 0...d.length)
 var c = haxe.zip.Uncompress.run(b);
 c.length == 0;
 #end
-#if php
+#if (php || ruby)
 var d = [120, 218, 43, 73, 45, 46, 1, 0, 4, 93, 1, 193];
 var b = haxe.io.Bytes.alloc(d.length);
 for (i in 0...d.length)
