--- a/haxe/ds/Vector.unit.hx
+++ b/haxe/ds/Vector.unit.hx
@@ -1,7 +1,7 @@
 var vec = new haxe.ds.Vector(3);
-var vNullInt = #if static 0 #else null #end;
-var vNullBool = #if static false #else null #end;
-var vNullFloat = #if static 0.0 #else null #end;
+var vNullInt = #if (static && !ruby) 0 #else null #end;
+var vNullBool = #if (static && !ruby) false #else null #end;
+var vNullFloat = #if (static && !ruby) 0.0 #else null #end;
 vec.length == 3;
 vec.get(0) == vNullInt;
 vec.get(1) == vNullInt;
